### PR TITLE
[fix] : fix internal ip ordering

### PR DESCRIPTION
**Pull request by rahulait**
### General:

* [ ] Have you removed all sensitive information, including but not limited to access keys and passwords?
* [ ] Have you checked to ensure there aren't other open or closed [Pull Requests](../../pulls) for the same bug/feature/question?

Node's internal ip is set based on whatever ip is returned as first internal ip. In case of VPC, we always want that ip to be node's internal ip. We do that ordering when returning node's ip-addresses. However, we were storing things in map and re-generating which was changing the order. Instead, we can just rely on map to tell if its unique or not and then append to the slice. This PR fixes that by making sure node's actual ip's are listed first and then the ones set by kubelet.

### Pull Request Guidelines:

1. [x] Does your submission pass tests?
1. [ ] Have you added tests? 
1. [ ] Are you addressing a single feature in this PR? 
1. [x] Are your commits atomic, addressing one change per commit?
1. [x] Are you following the conventions of the language? 
1. [x] Have you saved your large formatting changes for a different PR, so we can focus on your work?
1. [x] Have you explained your rationale for why this feature is needed? 
1. [ ] Have you linked your PR to an [open issue](https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/)

### CODE REVIEW

File: cloud/linode/instances.go [hunk 1]
```diff
@@ -284,12 +284,18 @@ func (i *instances) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloud
 		return nil, err
 	}
 
-	uniqueAddrs := make(map[string]v1.NodeAddressType, len(node.Status.Addresses)+len(ips))
+	addresses := []v1.NodeAddress{{Type: v1.NodeHostName, Address: linode.Label}}
 	for _, ip := range ips {
-		if _, ok := uniqueAddrs[ip.ip]; ok {
+		addresses = append(addresses, v1.NodeAddress{Type: ip.ipType, Address: ip.ip})
+	}
```
Comment (rahulait) [inline cloud/linode/instances.go:290]: Adding ips first here as the ordering is already done in getInstanceAddresses() https://github.com/linode/linode-cloud-controller-manager/blob/main/cloud/linode/instances.go#L41-L63

**Review by AshleyDumaine — APPROVED** [review]

File: cloud/linode/instances.go (hunk 1 above)
Comment (rosskirkpat) [reply cloud/linode/instances.go:290]: thanks for clarifying this, I was about to leave a comment asking about that 😄

File: cloud/linode/instances.go
[the remark this answers is not in the document]
Comment (rahulait) [reply cloud/linode/instances.go:1]: Not sure about first externalIP as ipv4 only as we never did that before, but let me see if we can add such logic easily to the code. Maybe remove external ipv6's from slice and add it to the end of slice.

[the remark this answers is not in the document]
Comment (rahulait) [reply cloud/linode/instances.go:1]: Yeah, true. Added test to make sure vpc ip is first and ipv6 at the end. external ipv4 and other internal ip's can be in any order.

**Review by rosskirkpat — APPROVED** [review]

**Pull request merged**